packages/edit-site/src/posts.js

```diff
@@ -11,16 +30,67 @@ import PostsApp from './components/posts-app';
 
 /**
  * Initializes the "Posts Dashboard"
- * @param {string} id DOM element id.
+ * @param {string} id       ID of the root element to render the screen in.
+ * @param {Object} settings Editor settings.
  */
-export function initializePostsDashboard( id ) {
+export function initializePostsDashboard( id, settings ) {
```

Comment:
Can we avoid using the settings from the server? I would love if we can fetch all the necessary settings using REST API instead.

Reply:
I'll explore this, yes. I'm wondering though why we have it like this in site editor for now..

Reply:
There's a lot of things that get through that settings object both in post and site editors and yeah initially the endpoints didn't have most of the settings. I'm fine with you did here for now. But let's track this. I think we should look at each setting separately and see how to avoid it.